docker-multinode/common.sh

```diff
@@ -0,0 +1,437 @@
+
+      yum -y -q install bridge-utils
+      ifconfig docker0 down
+      brctl delbr docker0 
+      systemctl restart docker
+      ;;
+    ubuntu|debian)
+      # Newer ubuntu and debian releases uses systemd. Handle that
+      if kube::helpers::command_exists systemctl; then
+        kube::multinode::restart_docker_systemd
+      else
+        DOCKER_CONF="/etc/default/docker"
+        
+        kube::helpers::file_replace_line ${DOCKER_CONF} \ # Replace content in this file
+          "--bip" \ # Find a line with this content...
+          "OPTIONS=\"\$OPTIONS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" # ...and replace the found line with this line
+
+        apt-get install -y bridge-utils 
+        brctl delbr docker0 
+        service docker stop
+        while [ $(ps aux | grep /usr/bin/docker | grep -v grep | wc -l) -gt 0 ]; do
+            kube::log::status "Waiting for docker to terminate"
+            sleep 1
+        done
+        service docker start
+      fi
+      ;;
+    systemd)
+      kube::multinode::restart_docker_systemd
+      ;;
+    *)
+        kube::log::error "Unsupported operations system ${lsb_dist}"
+        exit 1
+        ;;
+  esac
+
+  kube::log::status "Restarted docker with the new flannel settings"
+}
+
+# Replace --mtu and --bip in systemd's docker.service file and restart
+kube::multinode::restart_docker_systemd(){
+
+  DOCKER_CONF="/lib/systemd/system/docker.service"
+
+  # This expression checks if the "--mtu" and "--bip" options are there
+  # If they aren't, they are inserted at the end of the docker command
+  if [[ -z $(grep -- "--mtu=" $DOCKER_CONF) ]]; then
+    sed -e "s@$(grep "/usr/bin/docker" $DOCKER_CONF)@$(grep "/usr/bin/docker" $DOCKER_CONF) --mtu=${FLANNEL_MTU}@g" -i $DOCKER_CONF
+  fi
+  if [[ -z $(grep -- "--bip=" $DOCKER_CONF) ]]; then
+    sed -e "s@$(grep "/usr/bin/docker" $DOCKER_CONF)@$(grep "/usr/bin/docker" $DOCKER_CONF) --bip=${FLANNEL_SUBNET}@g" -i $DOCKER_CONF
+  fi
+
+  # Finds "--mtu=????" and replaces with "--mtu=${FLANNEL_MTU}"
+  # Also finds "--bip=??.??.??.??" and replaces with "--bip=${FLANNEL_SUBNET}"
+  sed -e "s@$(grep -o -- "--mtu=[[:graph:]]*" $DOCKER_CONF)@--mtu=${FLANNEL_MTU}@g;s@$(grep -o -- "--bip=[[:graph:]]*" $DOCKER_CONF)@--bip=${FLANNEL_SUBNET}@g" -i $DOCKER_CONF
+
+  systemctl daemon-reload
+  systemctl restart docker
+}
+
+# Start kubelet first and then the master components as pods
+kube::multinode::start_k8s_master() {
+  
+  kube::log::status "Launching Kubernetes master components..."
+
+  docker run -d \
+    --net=host \
+    --pid=host \
+    --privileged \
+    --restart=${RESTART_POLICY} \
+    ${KUBELET_MOUNTS} \
+    gcr.io/google_containers/hyperkube-${ARCH}:v${K8S_VERSION} \
+    /hyperkube kubelet \
+      --allow-privileged=true \
+      --api-servers=http://localhost:8080 \
+      --config=/etc/kubernetes/manifests-multi \
+      --cluster-dns=${DNS_SERVER_IP} \
+      --cluster-domain=${DNS_DOMAIN} \
+      --containerized \
```

Comment:
The [guide](http://kubernetes.io/docs/getting-started-guides/docker-multinode/master/) also has `--address=0.0.0.0 --enable-server --hostname-override=127.0.0.1` Which is right?
Also needs -v /var/log/containers/:/var/log/containers:rw to support logging